android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt

```diff
@@ -60,12 +60,12 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
     fun registerWith(registrar: Registrar) {
       Log.i(TAG, "registerWith")
 
-      val context = registrar.activity().applicationContext
-      val messenger = registrar.messenger()
-      HyperTrackPlugin().onAttachedToEngine(context, messenger)
+      registrar.activity()?.applicationContext?.let { context -> 
+        val messenger = registrar.messenger()
+        HyperTrackPlugin().onAttachedToEngine(context, messenger)
+      }
```

Comment:
we should log an error if activity is `null`
```suggestion
} ?: run {
  // log error if activity is null.
}
```